File: src/main/java/net/felsstudio/fels/Modules/mBox/mBox.java
package main.java.net.felsstudio.fels.Modules.mBox;

import javafx.scene.control.Alert;
import javafx.scene.control.ButtonType;
import main.java.net.felsstudio.fels.Modules.Module;
import main.java.net.felsstudio.fels.lib.*;
import javax.swing.JOptionPane;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Optional;

public class mBox implements Module {

    @Override
    public Map<String, Value> constants() {
        return Map.of();
    }

    @Override
    public Map<String, Function> functions() {
        final var result = new LinkedHashMap<String, Function>(5);

        result.put("showErrorMessage", args ->{
            JOptionPane.showMessageDialog(null, args[0].asString(),args[1].asString(),JOptionPane.ERROR_MESSAGE);
            return NumberValue.ZERO;
        });

        result.put("showInformationMessage",args ->{
            JOptionPane.showMessageDialog(null, args[0].asString(),args[1].asString(),JOptionPane.INFORMATION_MESSAGE);
            return NumberValue.ZERO;
        });

        result.put("showWarningMessage",args -> {
            JOptionPane.showMessageDialog(null, args[0].asString(),args[1].asString(),JOptionPane.WARNING_MESSAGE);
            return NumberValue.ZERO;
        });

        result.put("showNoneMessage",args -> {
            JOptionPane.showMessageDialog(null, args[0].asString(),args[1].asString(),JOptionPane.PLAIN_MESSAGE);
            return NumberValue.ZERO;
        });

        result.put("showInputDialog",args -> {
            return new StringValue(JOptionPane.showInputDialog(null,args[0], String.valueOf(args[1]),JOptionPane.QUESTION_MESSAGE));
        });

        return result;
    }
}

File: src/net/felsstudio/fels/Modules/mBox/mBox.kt
package net.felsstudio.fels.Modules.mBox;

import net.felsstudio.fels.lib.Functions;
import net.felsstudio.fels.lib.NumberValue;
import net.felsstudio.fels.lib.StringValue;
import net.felsstudio.fels.Modules.Module;

import javax.swing.*;

public class mBox implements Module {
    @Override
    public void init() {
        Functions.set("showErrorMessage", args ->{
            JOptionPane.showMessageDialog(null,args[0], String.valueOf(args[1]),JOptionPane.ERROR_MESSAGE);
            return NumberValue.ZERO;
        });
        Functions.set("showInformationMessage",args ->{
           JOptionPane.showMessageDialog(null,args[0], String.valueOf(args[1]),JOptionPane.INFORMATION_MESSAGE);
           return NumberValue.ZERO;
        });
        Functions.set("showQuestionMessage",args ->{
            JOptionPane.showMessageDialog(null,args[0], String.valueOf(args[1]),JOptionPane.QUESTION_MESSAGE);
            return NumberValue.ZERO;
        });
        Functions.set("showWarningMessage",args -> {
            JOptionPane.showMessageDialog(null,args[0], String.valueOf(args[1]),JOptionPane.WARNING_MESSAGE);
            return NumberValue.ZERO;
        });
        Functions.set("showPlanMessage",args -> {
            JOptionPane.showMessageDialog(null,args[0], String.valueOf(args[1]),JOptionPane.PLAIN_MESSAGE);
            return NumberValue.ZERO;
        });
        Functions.set("showInputDialog",args -> {
            return new StringValue(JOptionPane.showInputDialog(null,args[0], String.valueOf(args[1]),JOptionPane.QUESTION_MESSAGE));
        });
    }
}
